whylog/log_reader/searchers.py

```diff
@@ -68,3 +84,11 @@ def _reverse_from_offset(self, offset, buf_size=BufsizeConsts.STANDARD_BUF_SIZE)
                         yield line, actual_offset
             actual_offset = self._decrease_actual_offset_properly(actual_offset, truncated)
             yield truncated, actual_offset
+
+    def search(self, investigation_step):
```

Comment:
What do you suggests, it works fine with one file for entire investigation. But if we have many files to investigate, the offset provided by FrontInput is useful only for one file - the file containing the primary cause. So my code does not favour this one file and treats all files the same.